### PR TITLE
Transform Product Create/Update form into Live Component

### PR DESCRIPTION
**Pull request by jakubtobiasz**

### CODE REVIEW

File: behat.yml.dist [hunk 1]
```diff
@@ -26,7 +26,7 @@ default:
             files_path: "%paths.base%/src/Sylius/Behat/Resources/fixtures/"
             base_url: "https://127.0.0.1:8080/"
             default_session: symfony
-            javascript_session: panther
+            javascript_session: chromedriver
```
Comment (jakubtobiasz) [inline behat.yml.dist]: Something bad happened, and for unknown reason GitHub Actions can't run Chrome on the CI. Let's go with switching to the `chromedriver` for now, and trying to fix it later.

File: composer.json [hunk 2]
```diff
@@ -190,7 +191,8 @@
     "conflict": {
         "doctrine/orm": ">= 2.16.0",
         "lexik/jwt-authentication-bundle": "^2.18",
-        "stof/doctrine-extensions-bundle": "1.8.0"
+        "stof/doctrine-extensions-bundle": "1.8.0",
+        "symfony/ux-live-component": "2.15.0"
```
Comment (jakubtobiasz) [inline composer.json:195]: https://github.com/symfony/ux/issues/1547

File: features/admin/product/managing_products/editing_product_attributes.feature
Comment (jakubtobiasz) [inline features/admin/product/managing_products/editing_product_attributes.feature:1]: As I changed the way how the product attributes autocomplete works (it no longer keeps the added attributes) these scenarios became obsolete.

File: features/admin/product/managing_products/managing_attributes/adding_product_with_float_attribute.feature [hunk 3]
```diff
@@ -9,11 +9,12 @@ Feature: Adding a new product with a float attribute
         And the store has a non-translatable float product attribute "Display Size"
         And I am logged in as an administrator
 
-    @todo @ui @javascript @api
+    @ui @javascript @api
     Scenario: Adding a float attribute to a product
         When I want to create a new configurable product
         And I specify its code as "display_size"
         And I name it "Smartphone" in "English (United States)"
+        And I add the "Display Size" attribute to it
```
Comment (jakubtobiasz) [inline features/admin/product/managing_products/managing_attributes/adding_product_with_float_attribute.feature:17]: I had to split adding the attribute and setting its value

File: src/Sylius/Behat/Page/Admin/Product/CreateConfigurableProductPage.php [hunk 4]
```diff
@@ -104,20 +89,24 @@ public function getAttributeValidationErrors(string $attributeName, string $loca
...
-            'search' => '.ui.fluid.search.selection.dropdown',
-            'search_item_selected' => 'div.menu > div.item.selected',
-            'slug' => '#sylius_product_translations_en_US_slug',
-            'tab' => '.menu [data-tab="%name%"]',
-            'taxonomy' => 'a[data-tab="taxonomy"]',
-        ]);
+        return array_merge(
+            parent::getDefinedElements(),
+                [
+                'attribute' => '.attribute',
```
Comment (Wojdylak) [inline src/Sylius/Behat/Page/Admin/Product/CreateConfigurableProductPage.php]:
```suggestion
            parent::getDefinedElements(),
            [
                'attribute' => '.attribute',
```

File: src/Sylius/Behat/Page/Admin/Product/FormTrait.php [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+            'product_attribute_delete_button' => '[data-test-product-attribute-delete-button="%attributeName%"]',
+            'product_attribute_input' => 'input[name="product_attributes"]',
+            'product_attribute_tab' => '[data-test-product-attribute-tab="%name%"]',
+            'product_translation_accordion' => '[data-test-product-translation-accordion="%localeCode%"]',
+            'side_navigation_tab' => '[data-test-side-navigation-tab="%name%"]',
+        ];
+    }
+
+    /*
+     * Filling fields
```
Comment (Wojdylak) [inline src/Sylius/Behat/Page/Admin/Product/FormTrait.php:38]: Are these comments needed?

File: src/Sylius/Bundle/AdminBundle/TwigComponent/Product/FormComponent.php [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+use Symfony\UX\LiveComponent\Attribute\AsLiveComponent;
+use Symfony\UX\LiveComponent\Attribute\LiveAction;
+use Symfony\UX\LiveComponent\Attribute\LiveArg;
+use Symfony\UX\LiveComponent\Attribute\LiveProp;
+use Symfony\UX\LiveComponent\ComponentToolsTrait;
+use Symfony\UX\LiveComponent\DefaultActionTrait;
+use Symfony\UX\LiveComponent\LiveCollectionTrait;
+use Symfony\UX\TwigComponent\Attribute\ExposeInTemplate;
+
+#[AsLiveComponent(name: 'SyliusAdmin.Product.Form', template: '@SyliusAdmin/Product/_form.html.twig')]
```
Comment (Wojdylak) [inline src/Sylius/Bundle/AdminBundle/TwigComponent/Product/FormComponent.php:34]: Have we stopped keeping templates in the `Component` folder for LiveComponent?

Comment (jakubtobiasz) [reply src/Sylius/Bundle/AdminBundle/TwigComponent/Product/FormComponent.php:34]: IMO we should move away from putting components (at least the non-generic ones) into a bag of everything called `Component`. It doesn't matter whether it's a component or a regular template, as information about it comes from the config, so it's easy to be found.

File: src/Sylius/Behat/Page/Admin/Product/FormTrait.php (hunk 5 above)
Comment (jakubtobiasz) [reply src/Sylius/Behat/Page/Admin/Product/FormTrait.php:38]: Me and Janek found our `Create/Update` page classes for the class (multiplied by 2, as we have one set for configurable products, and another set for simple products). We'll slowly move next methods to this trait, so the sections (at least in my opinion) will help to find the place we need.

**Review by TheMilek — APPROVED** [review]

File: src/Sylius/Bundle/AdminBundle/Autocompleter/ProductAttributeAutocompleter.php [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+    {
+        return $this->productAttributeClass;
+    }
+
+    /**
+     * @param EntityRepository<ProductAttributeInterface> $repository
+     */
+    public function createFilteredQueryBuilder(EntityRepository $repository, string $query): QueryBuilder
+    {
+        $productAttributesToBeExcluded = $this->options['extra_options']['attributeCodes'];
```
Comment (Wojdylak) [inline src/Sylius/Bundle/AdminBundle/Autocompleter/ProductAttributeAutocompleter.php]:
```suggestion
        $productAttributesToBeExcluded = $this->options['extra_options']['attributeCodes'] ?? [];
```

File: src/Sylius/Bundle/AdminBundle/TwigComponent/Product/ProductAttributeAutocompleteComponent.php [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+            ]
+        ));
+    }
+
+    /**
+     * @return array<string>
+     */
+    public function hydrateSelectedAttributeCodes(string $value): array
+    {
+        return explode(',', $value);
```
Comment (Wojdylak) [inline src/Sylius/Bundle/AdminBundle/TwigComponent/Product/ProductAttributeAutocompleteComponent.php:66]:
```suggestion
        if ('' === $value) {
            return [];
        }

        return explode(',', $value);
```

**Review by Wojdylak — APPROVED** [review]

**Review by NoResponseMate — APPROVED** [review]

**Pull request merged**